sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DuplicateMetricStorageException.java

```diff
@@ -7,13 +7,30 @@
 
 import io.opentelemetry.sdk.metrics.internal.descriptor.MetricDescriptor;
 
-/** There are multiple metrics defined with the same name/identity. */
+/**
+ * There are multiple metrics defined with the same name/identity.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
 class DuplicateMetricStorageException extends IllegalArgumentException {
   private static final long serialVersionUID = 1547329629200005982L;
+  private final MetricDescriptor existing;
+  private final MetricDescriptor conflict;
 
   DuplicateMetricStorageException(
       MetricDescriptor existing, MetricDescriptor next, String message) {
     // TODO: Better error messages including async vs. sync instruments.
     super(message + " Found: " + existing + ", Want: " + next);
```

Comment:
I agree with the TODO here. I'm pretty sure I wouldn't understand what "Want" vs. "Found" would mean. :)

Reply:
Updated PTAL

Reply:
looks good!